### ISSUE TITLE
URLs with trailing slashes do not work (see note)

### ISSUE THREAD

**Issue by codyburleson** [codyburleson/cburleson-pwa#83]
The following URL shows up in Duck Duck Go search results:

https://codyburleson.com/copy-assets-from-node-modules-in-stencil-js/

The trailing slash does not work and it also does not return a Page Not Found (my internal error page).

Removing the trailing slash works. This may be having a bad impact on SEO.

**Comment by codyburleson** [comment]
The following JavaScript error appears in the browser console when attempting to use a trailing slash with one of the page or post URLs:

```
page-book-review-the-goldfinch.entry.js:28 Uncaught (in promise) TypeError: Cannot read property 'title' of undefined
    at PageBookReviewTheGoldfinch.componentWillLoad (page-book-review-the-goldfinch.entry.js:28)
    at safeCall (index-4c042e91.js:1518)
    at dispatchHooks (index-4c042e91.js:1302)
    at Array.dispatch (index-4c042e91.js:1284)
    at consumeTimeout (index-4c042e91.js:2734)
    at flush (index-4c042e91.js:2758)
```

`Cannot read property 'title' of undefined` indicates a problem in the `componentWillLoad()` method of each page component. The algorithm for deriving the page or post ID from the document path (URL) is not sufficient. It assumes there is no trailing slash and it uses a substring function to get text after the last trailing slash. Since the last trailing slash, in this case, has no text after it, the derived ID ends up being nothing (blank) and thus, the title of the document cannot be looked up by ID.

**Comment by codyburleson** [comment]
First, we must determine if the path ends with a trailing slash. If it does, then we need to remove it. An elegant and short way to get the last character of a string is by using the the `String.prototype.slice` method. A negative start index slices the string from length+index, to length, being index -1, the last character is extracted:

```
let path = document.location.pathname;
 // FIX FOR ISSUE #83...
if(path.slice(-1) === '/') { // If path ends in trailing slash,
       path = path.substring(0, path.length - 1); // then remove trailing slash from the path
}
```
Then, in the code that follows this snippet, we use the `path` variable rather than `document.location.pathname`

**Comment by codyburleson** [comment]
Resolved as per descriptions above.
